agent/engine/execcmd/agent_version.go

```diff
@@ -0,0 +1,64 @@
+package execcmd
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+)
+
+const (
+	versionDelimiter  = "."
+	badInputReturn    = -999
+	badFormatErrorStr = "invalid version format"
+)
+
+// compareAgentVersion compares two versions, returning -1 if v1 < v2, 0 if v1 == v2, and 1 if v1 > v2.
+// The version must have only integers representing major, minor, patch, and reserved; delimited by dots (e.g. "1.2.3.4").
+// An error is returned if any of the versions passed as parameter is malformed.
+// example:
+// SemverCompare("3.0.236.0", "3.0.237.0") returns -1, nil
```

Comment:
nit: comment is still referring to `SemverCompare`

Reply:
ah, thanks good catch